src/core/contextProxy.ts

```diff
@@ -0,0 +1,123 @@
+		return this.originalContext.extensionUri
+	}
+	get extensionPath(): string {
+		return this.originalContext.extensionPath
+	}
+	get globalStorageUri(): vscode.Uri {
+		return this.originalContext.globalStorageUri
+	}
+	get logUri(): vscode.Uri {
+		return this.originalContext.logUri
+	}
+	get extension(): vscode.Extension<any> | undefined {
+		return this.originalContext.extension
+	}
+	get extensionMode(): vscode.ExtensionMode {
+		return this.originalContext.extensionMode
+	}
+
+	// State management methods
+	async getGlobalState<T>(key: string): Promise<T | undefined>
+	async getGlobalState<T>(key: string, defaultValue: T): Promise<T>
+	async getGlobalState<T>(key: string, defaultValue?: T): Promise<T | undefined> {
+		// Check pending changes first
+		if (this.pendingStateChanges.has(key)) {
+			const value = this.pendingStateChanges.get(key) as T | undefined
+			return value !== undefined ? value : (defaultValue as T | undefined)
+		}
+		// Fall back to original context
+		return this.originalContext.globalState.get<T>(key, defaultValue as T)
+	}
+
+	async updateGlobalState<T>(key: string, value: T): Promise<void> {
+		if (this.disposed) {
+			throw new Error("Cannot update state on disposed context")
+		}
+		logger.debug(`ContextProxy: buffering state update for key "${key}"`)
+		this.pendingStateChanges.set(key, value)
+	}
+
+	// Secret storage methods
+	async getSecret(key: string): Promise<string | undefined> {
+		// Check pending changes first
+		if (this.pendingSecretChanges.has(key)) {
+			return this.pendingSecretChanges.get(key)
+		}
+		// Fall back to original context
+		return this.originalContext.secrets.get(key)
+	}
+
+	async storeSecret(key: string, value?: string): Promise<void> {
+		if (this.disposed) {
+			throw new Error("Cannot store secret on disposed context")
+		}
+		logger.debug(`ContextProxy: buffering secret update for key "${key}"`)
+		this.pendingSecretChanges.set(key, value)
+	}
+
+	// Save pending changes to actual context
+	async saveChanges(): Promise<void> {
+		if (this.disposed) {
+			throw new Error("Cannot save changes on disposed context")
+		}
+
+		// Apply state changes
+		if (this.pendingStateChanges.size > 0) {
+			logger.debug(`ContextProxy: applying ${this.pendingStateChanges.size} buffered state changes`)
+			for (const [key, value] of this.pendingStateChanges.entries()) {
+				await this.originalContext.globalState.update(key, value)
+			}
+			this.pendingStateChanges.clear()
+		}
+
+		// Apply secret changes
+		if (this.pendingSecretChanges.size > 0) {
+			logger.debug(`ContextProxy: applying ${this.pendingSecretChanges.size} buffered secret changes`)
+			for (const [key, value] of this.pendingSecretChanges.entries()) {
+				if (value === undefined) {
+					await this.originalContext.secrets.delete(key)
+				} else {
+					await this.originalContext.secrets.store(key, value)
+				}
+			}
+			this.pendingSecretChanges.clear()
+		}
+	}
+
+	// Called when the provider is disposing
```

Comment:
When does this happen?

Reply:
@mrubens  when cline provider is dispose, and cline provider dispose when webview dispose 

<img width="1053" alt="image" src="https://github.com/user-attachments/assets/e13fd470-faeb-4858-b15b-a5a60a4c9b59" />
<img width="847" alt="image" src="https://github.com/user-attachments/assets/50980d4f-434f-45e9-bba4-62df7a598c03" />